packages/vhd-lib/src/merge.js

```diff
@@ -21,111 +22,108 @@ export default limitConcurrency(2)(async function merge(
   childPath,
   { onProgress = noop } = {}
 ) {
-  const mergeStatePath = dirname(parentPath) + '/' + '.' + basename(parentPath) + '.merge.json'
-
-  const parentFd = await parentHandler.openFile(parentPath, 'r+')
-  try {
-    const parentVhd = new Vhd(parentHandler, parentFd)
-    const childFd = await childHandler.openFile(childPath, 'r')
-    try {
-      const childVhd = new Vhd(childHandler, childFd)
-
-      let mergeState = await parentHandler.readFile(mergeStatePath).catch(error => {
-        if (error.code !== 'ENOENT') {
-          throw error
+  await Disposable.use(
+    Disposable.all(
```

Comment:
I feel like `Disposable.all(` is not closed properly.

Reply:
fixed